node/bft/src/helpers/storage.rs

```diff
@@ -494,6 +494,49 @@ impl<N: Network> Storage<N> {
         Ok(missing_transmissions)
     }
 
+    /// Check the validity of a certificate coming from another validator.
+    ///
+    /// It suffices to check that the signers (author and endorsers) are members of the applicable committee
+    /// and that they form a quorum in the committee.
+    /// Under the fundamental fault tolerance assumption of at most `f` (stake of) faulty validators,
+    /// the quorum check on signers guarantees that at least one correct validator
+    /// has ensured the validity of the proposal contained in the certificate,
+    /// either by construction (by the author) or by checking (by an endorser):
+    /// given `N > 0` total stake, and `f` the largest integer `< N/3` (where `/` is exact rational division),
+    /// we have `N >= 3f + 1`, which implies `N - f >= 2f + 1`, which is always `> f`;
+    /// `N - f` is the quorum stake.
+    pub fn check_incoming_certificate(&self, certificate: &BatchCertificate<N>) -> Result<()> {
+        // Retrieve the certificate author and round.
+        let certificate_author = certificate.author();
+        let certificate_round = certificate.round();
+
+        // Retrieve the committee lookback.
+        let committee_lookback = self.ledger.get_committee_lookback_for_round(certificate_round)?;
+
+        // Ensure that the signers of the certificate reach the quorum threshold.
+        // Note that certificate.signatures() only returns the endorsing signatures, not the author's signature.
+        let mut signers: HashSet<Address<N>> =
+            certificate.signatures().map(|signature| signature.to_address()).collect();
+        signers.insert(certificate_author);
+        ensure!(
+            committee_lookback.is_quorum_threshold_reached(&signers),
```

Comment:
If we check the threshold here, do we still need to retrieve the`committee_lookback` in `process_batch_certificate_from_peer`?

Reply:
`process_batch_certificate` from peer uses the committee for a different purpose, namely deciding whether to advance the round.